scripts/conda-requirements.yaml

```diff
@@ -0,0 +1,120 @@
+channels:
+    - conda-forge
+    - ucb-bar
+    - nodefaults
+
+dependencies:
+    # https://conda-forge.org/feedstock-outputs/
+    #   filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute
+    #   instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax for constraining versions
+
+
+    # handy tool for introspecting package relationships and file ownership
+    # see https://github.com/rvalieris/conda-tree
+    - conda-tree
+
+    # bundle FireSim driver with deps into installer shell-script
+    - constructor
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    # elfutils has trouble building with gcc 11 for something that looks like it needs to be fixed upstream
+    # ebl_syscall_abi.c:37:64: error: argument 5 of type 'int *' declared as a pointer [-Werror=array-parameter=]
+    #    37 | ebl_syscall_abi (Ebl *ebl, int *sp, int *pc, int *callno, int *args)
+    #       |                                                           ~~~~~^~~~
+    # In file included from ./../libasm/libasm.h:35,
+    #                  from ./libeblP.h:33,
+    #                  from ebl_syscall_abi.c:33:
+    # ./libebl.h:254:46: note: previously declared as an array 'int[6]'
+    #   254 |                             int *callno, int args[6]);
+    #       |                                          ~~~~^~~~~~~
+    #
+    # pin to gcc=10 until we get that fixed.
+
+    - gcc =10 
+    - gxx =10 
+    - binutils 
+    - conda-gcc-specs
+
+    - dromajo
+    - riscv-tools
```

Comment:
I think we should install the tools separately in the `build-setup...` script from the `conda-requirements.yaml`. Either that or we remove the option in the setup script to use your own toolchain and we document that to choose another toolchain you have to uninstall `riscv-tools` from the env and install `esp-tools` (or use your own toolchain). I think the former option is a bit better (installing the toolchain separately) / less documentation.

Reply:
I was going to mention that best-practice is to create the conda env from a single list of requirements so that the solver can take all of the depencencies into account.  Installing things incrementally is possible but can be problematic.

I tried looking for a decent writeup of "why" it is best to install everything at once and I'm not finding it right now.  I'll look more later.

Reply:
I've thought it would probably be best to have things specified in multiple `conda-requirements.yaml` files that are appropriately collected by whatever helper script and given to a single `conda create env` command.  It can take multiple `-f` arguments to solve and install at once.

Reply:
I think that having multiple req. files would be fine. I just think there should be some sort of mechanism for users to choose which toolchain they want (probably by having the `build.sh` combine the main requirements + the toolchain requirements for them as you mentioned).